stackslib/src/config/mod.rs

```diff
@@ -2258,52 +2258,435 @@ impl Default for MinerConfig {
+    ///
+    /// Default: `9223372036854775807` (i64::MAX, effectively infinite, disabling automatic re-keying).
     pub private_key_lifetime: Option<u64>,
+    /// Target number of peers for StackerDB replication.
+    ///
+    /// Sets the maximum number of potential replication target peers requested from the
+    /// StackerDB control contract (`get-replication-targets`) when configuring a replica.
+    ///
+    /// Note: Formerly (pre-Epoch 3.0), this also controlled the target peer count for
+    /// inventory synchronization.
+    ///
+    /// Default: 32.
     pub num_neighbors: Option<u64>,
+    /// Maximum number of allowed concurrent inbound P2P connections.
+    ///
+    /// This acts as a hard limit. If the node already has this many active inbound P2P
+    /// connections, any new incoming P2P connection attempts will be rejected.
+    /// Outbound P2P connections initiated by this node are not counted against this limit.
+    ///
+    /// Default: 750.
     pub num_clients: Option<u64>,
+    /// Maximum total number of allowed concurrent HTTP connections.
+    ///
+    /// This limits the total number of simultaneous connections the node's RPC/HTTP server
+    /// will accept. If this limit is reached, new incoming HTTP connection attempts
+    /// will be rejected.
+    ///
+    /// Default: 1000.
     pub max_http_clients: Option<u64>,
+    /// Target number of outbound P2P connections the node aims to maintain.
+    ///
+    /// The connection pruning logic only activates if the current number of established
+    /// outbound P2P connections exceeds this value. Pruning aims to reduce the connection
+    /// count back down to this target, ensuring the node maintains a baseline number
+    /// of outbound peers for network connectivity.
+    ///
+    /// Default: 16.
     pub soft_num_neighbors: Option<u64>,
+    /// Soft limit threshold for triggering inbound P2P connection pruning.
+    ///
+    /// If the total number of currently active inbound P2P connections exceeds this value,
+    /// the node will activate pruning logic to reduce the count, typically by applying
+    /// per-host limits (see [`ConnectionOptionsFile::soft_max_clients_per_host`]).
+    /// This helps manage the overall load from inbound peers.
+    ///
+    /// Default: 750.
     pub soft_num_clients: Option<u64>,
+    /// maximum number of neighbors per host we permit
+    ///
+    /// Default: 1.
+    /// Deprecated: it does not have any effect on the node's behavior.
     pub max_neighbors_per_host: Option<u64>,
+    /// maximum number of inbound p2p connections per host we permit
+    ///
+    /// Default: 4.
+    /// Deprecated: it does not have any effect on the node's behavior.
     pub max_clients_per_host: Option<u64>,
+    /// soft limit on the number of neighbors per host we permit
+    ///
+    /// Default: 1.
+    /// Deprecated: it does not have any effect on the node's behavior.
```

Comment:
for consistency, are we certain this has never had an effect or possible it was deprecated for nakamoto (3.0) release?
the reason i ask is that i see later on a deprecation message specifically mentioning nakamoto, but this message implies it was deprecated before nakamoto (and if that is the case, perhaps it can be removed entirely).